cmudb/extensions/tscout/operating_unit_codegen.c

```diff
@@ -0,0 +1,44 @@
+/**
+ * @brief - An enum that represents all the C data types
+ * encountered while scanning the Postgres code-base.
+ *
+ */
+typedef enum c_type {
+  T_BOOL = 0,
+  T_SHORT,
+  T_INT,
+  T_LONG,
+  T_FLOAT,
+  T_DOUBLE,
+  T_ENUM,
+  T_PTR,
+  T_UNKNOWN,
+} c_type;
+
+/**
+ * @brief - A struct that represents an X or a feature.
+ * An X contains a name and a C data type.
+ *
+ */
+typedef struct field {
+  c_type type;
+  char *name;
+} field;
+
+/**
+ * @brief - Represents a TScout Operating Unit
```

Comment:
This is where being named differently from the actual TScout would be handy...

```suggestion
 * @brief - An OperatingUnit as defined by TScout.
```